Mathlib/Combinatorics/SimpleGraph/Matching.lean

```diff
@@ -133,6 +133,53 @@ lemma IsMatching.coeSubgraph {G' : Subgraph G} {M : Subgraph G'.coe} (hM : M.IsM
   · obtain ⟨_, hw', hvw⟩ := (coeSubgraph_adj _ _ _).mp hy
     rw [← hw.2 ⟨y, hw'⟩ hvw]
 
+lemma IsMatching.exists_of_disjoint_sets_of_equiv {s t : Set V} (h : Disjoint s t)
+    (f : s ≃ t) (hadj : ∀ {v}, (h : v ∈ s) → G.Adj v (f ⟨v, h⟩)) :
```

Comment:
What I had in mind was the rather more symmetric
```suggestion
    (f : s ≃ t) (hadj : ∀ v : s, G.Adj v (f v)) :
```
but your current spelling is perfectly fine